src/main/java/com/google/api/codegen/java/JavaSimpleFileProvider.java

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.java;
+
+import com.google.api.codegen.ApiConfig;
+import com.google.api.codegen.CodeGeneratorUtil;
+import com.google.api.codegen.GapicProvider;
+import com.google.api.codegen.GeneratedResult;
+import com.google.api.codegen.InputElementView;
+import com.google.api.codegen.SnippetDescriptor;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.Multimap;
+
+import java.io.IOException;
+
+/**
+ * The GapicProvider which runs Gapic code generation for Java.
+ */
+public class JavaSimpleFileProvider<T> implements GapicProvider<Iterable<T>> {
```

Comment:
This file isn't driving anything to be "simpler"; the only difference between it and JavaGapicProvider is that its element is an Iterable and the JavaGapicProvider element is a single element. Maybe "JavaIterableProvider."